pkg/apis/core/validation/validation.go

```diff
@@ -5853,8 +5864,42 @@ func ValidateContainerStateTransition(newStatuses, oldStatuses []core.ContainerS
 }
 
 // ValidateInitContainerStateTransition test to if any illegal init container state transitions are being attempted
-func ValidateInitContainerStateTransition(newStatuses, oldStatuses []core.ContainerStatus, fldpath *field.Path, podSpec *core.PodSpec) field.ErrorList {
+func ValidateInitContainerStateTransition(newStatuses, oldStatuses []core.ContainerStatus, fldpath *field.Path, podSpec core.PodSpec) field.ErrorList {
 	allErrs := field.ErrorList{}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ContainerRestartRules) {
```

Comment:
We usually don't want to be checking gates directly inside validation.  We generally check gates in strategy and pass options.  In this case, do you even need to check the gate?  If you get here and the field is set, it is by definition allowed - just validate it